src/app/components/projects/pages/details/details.component.ts

```diff
@@ -51,28 +56,109 @@ const projectKey = "project";
+      ></baw-debounce-input>
+
+      <baw-loading [display]="loading"></baw-loading>
+
+      <p *ngIf="!hasSites && !hasRegions && !loading" class="lead">
+        No additional data to display here, try adding sites to the project
       </p>
 
-      <app-site-cards *ngIf="hasSites" [project]="project"></app-site-cards>
-      <app-region-cards
-        *ngIf="hasRegions"
-        [project]="project"
-      ></app-region-cards>
+      <div id="model-grid">
+        <!-- Google Maps -->
+        <div *ngIf="hasSites || hasRegions" class="item map">
+          <app-site-map [project]="project"></app-site-map>
+        </div>
+
+        <!-- Regions -->
+        <div *ngFor="let region of regions" class="item">
+          <app-site-card [project]="project" [region]="region"></app-site-card>
+        </div>
+
+        <!-- Sites -->
+        <div *ngFor="let site of sites" class="item">
+          <app-site-card [project]="project" [site]="site"></app-site-card>
+        </div>
+      </div>
+
+      <ngb-pagination
+        *ngIf="displayPagination"
+        aria-label="Pagination Buttons"
+        class="mt-2 d-flex justify-content-end"
+        [collectionSize]="collectionSize"
+        [(page)]="page"
+      ></ngb-pagination>
     </ng-container>
   `,
   styleUrls: ["./details.component.scss"],
 })
-class DetailsComponent extends PageComponent implements OnInit {
+class DetailsComponent extends PaginationTemplate<any> implements OnInit {
+  public collectionSize = 0;
+  public collectionSizes = { sites: 0, regions: 0 };
   public defaultDescription = "<i>No description found</i>";
-  public project: Project;
   public hasRegions: boolean;
   public hasSites: boolean;
+  public project: Project;
+  public regions: List<Region>;
+  public sites: List<Site>;
+  /**
+   * Api requests are independent, use this to track when both complete
+   */
+  private apiReturnCount = 0;
+
+  constructor(
+    route: ActivatedRoute,
+    router: Router,
+    config: NgbPaginationConfig,
+    private regionsApi: RegionsService,
+    private sitesApi: SitesService
+  ) {
+    super(
+      router,
+      route,
+      config,
+      undefined,
+      "name",
+      () => [this.project.id],
+      (models) => {
+        this.apiReturnCount++;
+        this.loading = this.apiReturnCount !== 2;
+
+        if (models.length === 0) {
+          return;
+        }
 
-  constructor(private route: ActivatedRoute) {
-    super();
+        const collectionSize = models[0].getMetadata().paging.total || 0;
+
+        if (models[0] instanceof Site) {
+          this.sites = List(models);
+          this.collectionSizes.sites = collectionSize;
+          this.hasSites = true;
+        } else {
+          this.regions = List(models);
+          this.collectionSizes.regions = collectionSize;
+          this.hasRegions = true;
+        }
+
+        this.collectionSize = Math.max(
+          this.collectionSizes.sites,
+          this.collectionSizes.regions
+        );
```

Comment:
what is this doing?

Reply:
Trying to find which list of models has the greater maximum page number